File: version/1.8.9/src/main/java/com/github/sorusclient/client/adapter/v1_8_9/RenderUtil.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter.v1_8_9;

import lombok.val;
import v1_8_9.net.minecraft.client.render.Tessellator;
import v1_8_9.net.minecraft.client.render.VertexFormats;

public class RenderUtil {

    public static void drawTexture(int x, int y, double textureX, double textureY, int width, int height) {
        textureX += 0.1;

        val tessellator = Tessellator.getInstance();
        val bufferBuilder = tessellator.getBuffer();

        bufferBuilder.begin(7, VertexFormats.POSITION_TEXTURE);
        bufferBuilder.vertex(x, y + height, 0.0).texture(textureX / 256, ((textureY + height) / 256)).next();
        bufferBuilder.vertex(x + width, y + height, 0.0).texture(((textureX + width) / 256), ((textureY + height) / 256)).next();
        bufferBuilder.vertex(x + width, y, 0.0).texture(((textureX + width) / 256), ((textureY + 0) / 256)).next();
        bufferBuilder.vertex(x, y, 0.0).texture(((textureX + 0) / 256), ((textureY + 0) / 256)).next();

        tessellator.draw();
    }

}

File: version/1.8.9/src/main/kotlin/com/github/sorusclient/client/adapter/v1_8_9/RenderUtil.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter.v1_8_9

import v1_8_9.net.minecraft.client.render.Tessellator
import v1_8_9.net.minecraft.client.render.VertexFormats

fun drawTexture(var1: Int, var2: Int, var3: Double, var4: Double, var5: Int, var6: Int) {
    val var3 = var3 + 0.1
    val var9 = Tessellator.getInstance()
    val var10 = var9.buffer
    var10.begin(7, VertexFormats.POSITION_TEXTURE)
    var10.vertex((var1 + 0).toDouble(), (var2 + var6).toDouble(), 0.0).texture(((var3 + 0).toFloat() / 256).toDouble(), ((var4 + var6).toFloat() / 256).toDouble()).next()
    var10.vertex((var1 + var5).toDouble(), (var2 + var6).toDouble(), 0.0).texture(((var3 + var5).toFloat() / 256).toDouble(), ((var4 + var6).toFloat() / 256).toDouble()).next()
    var10.vertex((var1 + var5).toDouble(), (var2 + 0).toDouble(), 0.0).texture(((var3 + var5).toFloat() / 256).toDouble(), ((var4 + 0).toFloat() / 256).toDouble()).next()
    var10.vertex((var1 + 0).toDouble(), (var2 + 0).toDouble(), 0.0).texture(((var3 + 0).toFloat() / 256).toDouble(), ((var4 + 0).toFloat() / 256).toDouble()).next()
    var9.draw()
}